src/contexts/wallet.context.tsx

```diff
@@ -0,0 +1,70 @@
+import { createContext, PropsWithChildren, useContext, useEffect } from 'react';
+import Web3 from 'web3';
+import { useApi } from '../hooks/api.hook';
+import { useSession } from '../hooks/session.hook';
+
+interface WalletInterface {
+  address?: string;
+  login: () => Promise<void>;
+  isInstalled: boolean;
+}
+
+const WalletContext = createContext<WalletInterface>(undefined as any);
+
+export function useWalletContext(): WalletInterface {
+  return useContext(WalletContext);
+}
+
+export function WalletContextProvider(props: PropsWithChildren): JSX.Element {
+  const { address, setAddress } = useSession();
+  const { getSignMessage } = useApi();
+  const { ethereum } = window as any;
```

Comment:
What is this? 😆

Reply:
as far as I have understood, metamask browser exchanges makes an ethereum object available on window. This should be used to hook into some functionality. `Web3` additionally is a wrapper, which makes a few things easier, more accessible and typed for better TS support